syntax = 'proto3';

// You can ignore these for now
//option java_multiple_files = true;
//option java_package = "example-digestor.resource.grpc.digestor";
//option java_outer_classname = "DigestorProto";
//option objc_class_prefix = "DIGEST";

package digestor;

service Digestor{
    // 定义一个 gRPC 服务方法， 在此也可以定义多个方法
    // rpc DoSomething(ClassType1) returns (ClassType2) {};
    rpc GetDigestor(DigestMessage) returns (DigestedMessage) {};
}


message DigestMessage{
    string ToDigest = 1;
}

message DigestedMessage{
    string Digested = 1;
    bool WasDigest = 2;
}
